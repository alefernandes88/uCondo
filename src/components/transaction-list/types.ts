import type { Account } from '../../types/account'

interface TransactionCallbacks {
  onErase: (account: Account) => void
  onSelect: (account: Account) => void
}

export interface TransactionListItemProps extends TransactionCallbacks {
  account: Account
}

export interface TransactionListProps extends TransactionCallbacks {
  data: Account[]
}
