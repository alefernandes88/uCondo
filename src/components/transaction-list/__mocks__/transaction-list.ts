import type { Account } from '../../../types/account'

export const mockOptions: Account[] = [
  {
    parentAccount: '',
    code: '1',
    name: 'Parent',
    transactionType: 'revenue',
    hasIncoming: false
  },
  {
    parentAccount: '',
    code: '2',
    name: 'Parent expenses',
    transactionType: 'expenses',
    hasIncoming: false
  },
  {
    parentAccount: '2',
    code: '2.1',
    name: 'Child expenses',
    transactionType: 'revenue',
    hasIncoming: true
  }
]
