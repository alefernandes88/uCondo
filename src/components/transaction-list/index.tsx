import React from 'react'
import { TouchableNativeFeedback } from 'react-native'
import TrashIcon from '../../icons/trash'
import { usePluralize } from '../../hook/pluralize'
import type { Account } from '../../types/account'
import {
  ContainerList,
  ContainerTitleWrapper,
  CounterLabel, ItemAction, ItemActionWrapper,
  ItemContainer,
  ItemText,
  ListContainer,
  ListContent,
  Title
} from './styles'
import type { TransactionListItemProps, TransactionListProps } from './types'
const listRenderItem = ({ account, onErase, onSelect }: TransactionListItemProps) => (
    <TransactionList.Item
        account={account}
        onErase={onErase}
        onSelect={onSelect}
    />
)

const TransactionList = ({
  data,
  onErase,
  onSelect
}: TransactionListProps) => {
  const dataCounter = data.length
  const counterLabel = usePluralize(dataCounter, 'Registro', 's')
  return (
        <ListContainer>
            <ContainerTitleWrapper>
                <Title>
                    Listagem
                </Title>
                <CounterLabel>
                    {dataCounter} {counterLabel}
                </CounterLabel>
            </ContainerTitleWrapper>

            <ContainerList
                data={data}
                renderItem={({ item }: { item: Account }) => listRenderItem({ account: item, onErase, onSelect })}
                keyExtractor={(item: Account) => item?.code}
            />
        </ListContainer>
  )
}

TransactionList.Item = ({
  account,
  onErase,
  onSelect
}: TransactionListItemProps) => {
  const {
    code,
    name,
    transactionType
  } = account
  return (
        <ItemContainer>
          <TouchableNativeFeedback
            testID={`transaction-item-${code}`}
            onPress={() => { onSelect(account) }}
            background={TouchableNativeFeedback.Ripple('#e4e4e4', true)}
          >
            <ListContent>
              <ItemText type={transactionType} numberOfLines={1}>
                {code} - {name}
              </ItemText>
              <ItemActionWrapper>
                {typeof onErase === 'function'
                  ? (
                    <TouchableNativeFeedback
                      testID={`transaction-item-${code}-erase`}
                      onPress={() => { onErase(account) }}
                      background={TouchableNativeFeedback.Ripple('#e4e4e4', true)}
                    >
                      <ItemAction>
                        <TrashIcon />
                      </ItemAction>
                    </TouchableNativeFeedback>
                    )
                  : null}
              </ItemActionWrapper>
            </ListContent>
          </TouchableNativeFeedback>
        </ItemContainer>
  )
}

export default TransactionList
