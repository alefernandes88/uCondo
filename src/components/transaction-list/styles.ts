import styled from 'styled-components/native'
import { FlatList } from 'react-native'
import type { AccountTransactionType, Account } from '../../types/account'

export const ItemContainer = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  border-radius: 15px;
  margin-bottom: 15px;
`

export const ItemText = styled.Text<{
  type: AccountTransactionType
}>`
  font-family: "Rubik_400Regular";
  font-size: 15px;
  color: ${({ type }) => type === 'revenue' ? '#1BA803' : '#E28856'};
  flex: 1;
  padding-right: 5px;
`

export const ContainerList = styled(FlatList<Account>)`
  padding-bottom: 55px;
  padding-top: 10px;
  border-top-color: rgba(148, 148, 148, 0.3);
  border-top-style: solid;
  border-top-width: 1px;
`

export const ContainerTitleWrapper = styled.View`
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 18px;
`

export const Title = styled.Text`
  font-family: "Rubik_400Regular";
  font-size: 20px;
  color: #3D3D4C;
`

export const CounterLabel = styled.Text`
  font-family: Roboto;
  font-size: 15px;
  color: #A0A0B2;
`

export const ListContainer = styled.View`
  flex: 1;
`

export const ItemAction = styled.View`
  padding: 5px;
`
export const ItemActionWrapper = styled.View`
  border-radius: 15px;
`

export const ListContent = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 10px 15px;
`
