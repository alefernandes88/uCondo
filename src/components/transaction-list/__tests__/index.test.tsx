import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { mockOptions } from '../__mocks__/transaction-list'
import TransactionList from '../index'

describe('Components -> Transaction List', () => {
  const onEraseSpy = jest.fn()
  const onSelectSpy = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should render with a normal style', () => {
    const { toJSON } = render((
      <TransactionList
        data={mockOptions}
        onErase={onEraseSpy}
        onSelect={onSelectSpy}
      />
    ))

    expect(toJSON()).toMatchSnapshot()
  })

  it('should select the item', async () => {
    const { findByTestId } = render((
      <TransactionList
        data={mockOptions}
        onErase={onEraseSpy}
        onSelect={onSelectSpy}
      />
    ))

    const item = await findByTestId('transaction-item-1')
    fireEvent.press(item)
    expect(onSelectSpy).toBeCalledWith(mockOptions[0])
  })

  it('should erase the item', async () => {
    const { findByTestId } = render((
      <TransactionList
        data={mockOptions}
        onErase={onEraseSpy}
        onSelect={onSelectSpy}
      />
    ))

    const item = await findByTestId('transaction-item-2.1-erase')
    fireEvent.press(item)
    expect(onEraseSpy).toBeCalledWith(mockOptions[2])
  })
})
