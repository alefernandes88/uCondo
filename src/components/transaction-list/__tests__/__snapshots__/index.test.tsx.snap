// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Transaction List should render with a normal style 1`] = `
<View
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 18,
      }
    }
  >
    <Text
      style={
        {
          "color": "#3D3D4C",
          "fontFamily": "Rubik_400Regular",
          "fontSize": 20,
        }
      }
    >
      Listagem
    </Text>
    <Text
      style={
        {
          "color": "#A0A0B2",
          "fontFamily": "Roboto",
          "fontSize": 15,
        }
      }
    >
      3
       
      Registros
    </Text>
  </View>
  <RCTScrollView
    data={
      [
        {
          "code": "1",
          "hasIncoming": false,
          "name": "Parent",
          "parentAccount": "",
          "transactionType": "revenue",
        },
        {
          "code": "2",
          "hasIncoming": false,
          "name": "Parent expenses",
          "parentAccount": "",
          "transactionType": "expenses",
        },
        {
          "code": "2.1",
          "hasIncoming": true,
          "name": "Child expenses",
          "parentAccount": "2",
          "transactionType": "revenue",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "borderTopColor": "rgba(148, 148, 148, 0.3)",
        "borderTopStyle": "solid",
        "borderTopWidth": 1,
        "paddingBottom": 55,
        "paddingTop": 10,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 15,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            testID="transaction-item-1"
          >
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#1BA803",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 15,
                  "paddingRight": 5,
                }
              }
              type="revenue"
            >
              1
               - 
              Parent
            </Text>
            <View
              style={
                {
                  "borderBottomLeftRadius": 15,
                  "borderBottomRightRadius": 15,
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  }
                }
                testID="transaction-item-1-erase"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="20"
                  bbWidth="20"
                  fill="none"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={20}
                  vbWidth={20}
                  width={20}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M2.5 5H4.16667H17.5"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291085521,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="1.5"
                    />
                    <RNSVGPath
                      d="M6.66668 5.00002V3.33335C6.66668 2.89133 6.84227 2.4674 7.15484 2.15484C7.4674 1.84228 7.89132 1.66669 8.33335 1.66669H11.6667C12.1087 1.66669 12.5326 1.84228 12.8452 2.15484C13.1578 2.4674 13.3333 2.89133 13.3333 3.33335V5.00002M15.8333 5.00002V16.6667C15.8333 17.1087 15.6578 17.5326 15.3452 17.8452C15.0326 18.1578 14.6087 18.3334 14.1667 18.3334H5.83335C5.39132 18.3334 4.9674 18.1578 4.65483 17.8452C4.34227 17.5326 4.16668 17.1087 4.16668 16.6667V5.00002H15.8333Z"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291085521,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="1.5"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 15,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            testID="transaction-item-2"
          >
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#E28856",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 15,
                  "paddingRight": 5,
                }
              }
              type="expenses"
            >
              2
               - 
              Parent expenses
            </Text>
            <View
              style={
                {
                  "borderBottomLeftRadius": 15,
                  "borderBottomRightRadius": 15,
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  }
                }
                testID="transaction-item-2-erase"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="20"
                  bbWidth="20"
                  fill="none"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={20}
                  vbWidth={20}
                  width={20}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M2.5 5H4.16667H17.5"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291085521,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="1.5"
                    />
                    <RNSVGPath
                      d="M6.66668 5.00002V3.33335C6.66668 2.89133 6.84227 2.4674 7.15484 2.15484C7.4674 1.84228 7.89132 1.66669 8.33335 1.66669H11.6667C12.1087 1.66669 12.5326 1.84228 12.8452 2.15484C13.1578 2.4674 13.3333 2.89133 13.3333 3.33335V5.00002M15.8333 5.00002V16.6667C15.8333 17.1087 15.6578 17.5326 15.3452 17.8452C15.0326 18.1578 14.6087 18.3334 14.1667 18.3334H5.83335C5.39132 18.3334 4.9674 18.1578 4.65483 17.8452C4.34227 17.5326 4.16668 17.1087 4.16668 16.6667V5.00002H15.8333Z"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291085521,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="1.5"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 15,
              "borderBottomRightRadius": 15,
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 15,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            testID="transaction-item-2.1"
          >
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#1BA803",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 15,
                  "paddingRight": 5,
                }
              }
              type="revenue"
            >
              2.1
               - 
              Child expenses
            </Text>
            <View
              style={
                {
                  "borderBottomLeftRadius": 15,
                  "borderBottomRightRadius": 15,
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "paddingBottom": 5,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                    "paddingTop": 5,
                  }
                }
                testID="transaction-item-2.1-erase"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="20"
                  bbWidth="20"
                  fill="none"
                  focusable={false}
                  height={20}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  vbHeight={20}
                  vbWidth={20}
                  width={20}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M2.5 5H4.16667H17.5"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291085521,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="1.5"
                    />
                    <RNSVGPath
                      d="M6.66668 5.00002V3.33335C6.66668 2.89133 6.84227 2.4674 7.15484 2.15484C7.4674 1.84228 7.89132 1.66669 8.33335 1.66669H11.6667C12.1087 1.66669 12.5326 1.84228 12.8452 2.15484C13.1578 2.4674 13.3333 2.89133 13.3333 3.33335V5.00002M15.8333 5.00002V16.6667C15.8333 17.1087 15.6578 17.5326 15.3452 17.8452C15.0326 18.1578 14.6087 18.3334 14.1667 18.3334H5.83335C5.39132 18.3334 4.9674 18.1578 4.65483 17.8452C4.34227 17.5326 4.16668 17.1087 4.16668 16.6667V5.00002H15.8333Z"
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291085521,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="1.5"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
